api/v1alpha1/openapi.yaml

```diff
@@ -2181,6 +2181,141 @@ components:
+    ConfigHookAction:
+      type: object
+      oneOf:
+        - $ref: '#/components/schemas/ConfigHookActionSystemdSpec'
+        - $ref: '#/components/schemas/ConfigHookActionExecutableSpec'
+      discriminator:
+        propertyName: type
+        mapping:
+          Systemd: '#/components/schemas/ConfigHookActionSystemdSpec'
+          Executable: '#/components/schemas/ConfigHookActionExecutableSpec'
+    ConfigHookActionSpec:
+      type: object
+      properties:
+        type:
+          type: string
+        triggerOn:
+          type: array
+          items:
+            $ref: '#/components/schemas/FileOperation'
+        timeout:
+             type: string
+             pattern: '^[1-9]\d*[smhd]$'
+             description: "Duration after which the action will be terminated. Format: number followed by 's' for seconds, 'm' for minutes, 'h' for hours, 'd' for days. Must be a positive integer."
+      required:
+        - type
+        - triggerOn
+    ConfigHookActionSystemdSpec:
+       allOf:
+        - $ref: '#/components/schemas/ConfigHookActionSpec'
+        - type: object
+          properties:
+            unit:
+              $ref: '#/components/schemas/ConfigHookActionSystemdUnit'
+          discriminator:
+            propertyName: type
+          required:
+            - unit
+    ConfigHookActionSystemdUnit:
+      type: object
+      properties:
+        name:
+          type: string
+          description: 'The name of the systemd unit on which the specified operations will be performed. This should be the exact name of the unit file, such as example.service. If the name is not populated the name will be auto discovered from the file path.'
+        operations:
+          type: array
+          items:
+            type: string
+            enum:
+              - "Enable"
+              - "Disable"
+              - "Start"
+              - "Stop"
+              - "Restart"
+              - "Reload"
+              - "DaemonReload"
+            x-enum-varnames:
+              - "SystemdEnable"
+              - "SystemdDisable"
+              - "SystemdStart"
+              - "SystemdStop"
+              - "SystemdRestart"
+              - "SystemdReload"
+              - "SystemdDaemonReload"
+          description: 'The specific systemd operations to perform on the specified unit.'
+        workDir:
+            type: string
+            description: 'The directory in which the executable will be run from if it is left empty it will run from the users home directory.'
+      required:
+        - name
+        - operations 
+    ConfigHookActionExecutableSpec:
+      allOf:
+        - $ref: '#/components/schemas/ConfigHookActionSpec'
+        - type: object
+          properties:
+            executable:
+              $ref: '#/components/schemas/ConfigHookActionExecutable'
+          discriminator:
+            propertyName: type
+          required:
+            - executable
+    ConfigHookActionExecutable:
```

Comment:
Reuse this for applications so make more generic.

Reply:
Maybe we can add the changed file as an argument too?

Reply:
Yeah I have been playing with how this will look my current approach is in the example.